src/Providers/AuthUserProvider.tsx

```diff
@@ -55,9 +55,7 @@ export default function AuthUserProvider({ children, unauthorized }: Props) {
     localStorage.removeItem(LocalStorageKeys.refreshToken);
 
     await refetch();
-
-    const redirectURL = getRedirectURL();
-    navigate(redirectURL ? `/?redirect=${redirectURL}` : "/");
+    navigate(getRedirectOr(`/login?redirect=${location.href}`));
```

Comment:
This behaviour is not needed when we click sign-out button right? Because the user voluntarily signed out.

Take a boolean param as shown instead:

```py
signOut(noRedirect=True) # skips adding redirect as qparam.
signOut() # adds redirect param.
```